sdk/src/Services/S3/Custom/_async/AmazonS3Client.Extensions.cs

```diff
@@ -145,7 +145,21 @@ Task ICoreAmazonS3.MakeObjectPublicAsync(string bucket, string objectKey, bool e
 
         Task ICoreAmazonS3.EnsureBucketExistsAsync(string bucketName)
         {
-            return this.PutBucketAsync(bucketName);
+            Task task = this.PutBucketAsync(bucketName);
```

Comment:
You should just use the async/await something like.

```
async Task ICoreAmazonS3.EnsureBucketExistsAsync(string bucketName)
{
   try
   {
        await this.PutBucketAsync(bucketName);
   }
   catch(BucketAlreadyOwnedByYouException)
   {
   }
}

Reply:
Fixed. It's working. Starting dry-run.